### PR TITLE
Remove obsolete Backported classes for jackson 1.5 support

### PR DESCRIPTION
**Pull request by asalamon74**
### Description

This ancient commit 10 years ago ( https://github.com/opensearch-project/opensearch-hadoop/commit/b919c173d4ad1dabc940c8edc40a6ebf0c9c0d05 ) added Backported classes to support jackson 1.5

But the hadoop version was quickly updated back to 1.8.8: https://github.com/opensearch-project/opensearch-hadoop/commit/76ca1a70b2458a031c545490c202370d1bc3d64d

And the hadoop version in the project are not using 1.5 anymore:

hadoop 3.1.2: jackson 1.9.13
hadoop 2.7.6: jackson 1.9.13
hadoop 2.2.0: jackson 1.8.8

so I think that jackson 1.5 support is no longer imporant.

These hardwired classes make it difficult to bump the jackson version to 1.9.13

### Issues Resolved

This PR removes the backported classes and makes it easier to bump the jackson version later.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 ### Removed
 
 ### Fixed
+- Remove obsolete Backported classes for jackson 1.5 support
```
Comment (Xtansia) [inline CHANGELOG.md:15]: This should be under `### Removed` and include the PR link at the end of the line, additionally fix casing of backported and Jackson:
```suggestion
- Removed obsolete backported classes for Jackson 1.5 support
```

**Review by Xtansia — CHANGES_REQUESTED** [review]
Thanks for the contribution @asalamon74! Changes look reasonable to me, just need to adjust the changelog entry a bit

**Comment by asalamon74** [comment]
Thanks for the review @Xtansia I've updated the changelog.

**Review by Xtansia — APPROVED** [review]
Thank you @asalamon74, really appreciate your efforts!

**Pull request merged**